swiftui/PasswordlessSwiftUI/PasswordlessSwiftUI/ContentView.swift

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import Firebase
+
+/// Main view where user can login in using Email Link Authentication.
+struct ContentView: View {
+  @State private var email: String = ""
+  @State private var isPresentingSheet = false
+
+  var body: some View {
+    NavigationView {
+      VStack(alignment: .leading) {
+        Text("Authenticate users with only their email, no password required!")
+          .padding(.bottom, 60)
+
+        CustomStyledTextField(
+          text: $email, placeholder: "Email", symbolName: "person.circle.fill"
+        )
+
+        CustomStyledButton(title: "Send Sign In Link", action: sendSignInLink)
+          .disabled(email.isEmpty)
+
+        Spacer()
+      }
+      .padding()
+      .navigationBarTitle("Passwordless Login")
+    }
+    .onOpenURL { url in
+      let link = url.absoluteString
+      if Auth.auth().isSignIn(withEmailLink: link) {
+        passwordlessSignIn(email: email, link: link) { result in
+          switch result {
+          case let .success(user):
+            isPresentingSheet = user?.isEmailVerified ?? false
+          case .failure:
+            isPresentingSheet = false
+          }
+        }
+      }
+    }
+    .sheet(isPresented: $isPresentingSheet) {
+      SuccessView(email: email)
+    }
+  }
+
+  // MARK: - Firebase 🔥
+
+  private func sendSignInLink() {
+    let actionCodeSettings = ActionCodeSettings()
+    actionCodeSettings.url = URL(string: "https://passwordlessswiftui.page.link/demo_login")
```

Comment:
I wonder if it might be a good idea to extract project-dependent constants and move them to a configuration struct that developers can then update with their project-specific values.

Reply:
As far as project constants, I can think of the above URL and some of the button titles, textfield placeholders, etc. Are these items I should move to a configuration struct?

Reply:
I was thinking mostly of the URL, which would also benefit from being decomposed into the prefix and short link. Button titles etc. should be dealt with proper i18n tools (which might be beyond the scope of this sample, but I'll leave it to @ryanwilson to comment on how stringent we are with i18n in our code samples).

Reply:
Yep I think we can leave the i18n out for this - breaking the project constant out to a separate file SGTM, no other variables are needed there.